src/main/webapp/src/app/recommendation/recommendation-list/recommendation-list.component.html

```diff
@@ -1,3 +1,4 @@
+<mat-spinner *ngIf="!hiddenGems"></mat-spinner>
```

Comment:
hiddenGems is set to an empty list on construction so hidden gems will always be true. 
I would write a function that checks if hidden gems is an array and if it has anything in it, and call that here

Reply:
like hiddenGems != null ?

Reply:
No so notice how in recommentation-list.component.ts has hiddenGems = [];
That basically means it's never going to be null, so you're never going to see the spinner even though things haven't loaded.
You can add a function to the ts class that does a more complicated check and returns a boolean, then you can call that from the html file in the ngif.